plugins/package-curation-providers/sw360/src/main/kotlin/Sw360PackageCurationProvider.kt

```diff
@@ -39,41 +39,74 @@ import org.ossreviewtoolkit.model.Package
 import org.ossreviewtoolkit.model.PackageCuration
 import org.ossreviewtoolkit.model.PackageCurationData
 import org.ossreviewtoolkit.model.RemoteArtifact
-import org.ossreviewtoolkit.model.config.Sw360StorageConfiguration
 import org.ossreviewtoolkit.model.jsonMapper
 import org.ossreviewtoolkit.model.orEmpty
+import org.ossreviewtoolkit.plugins.api.OrtPlugin
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
+import org.ossreviewtoolkit.plugins.api.Secret
 import org.ossreviewtoolkit.plugins.packagecurationproviders.api.PackageCurationProvider
 import org.ossreviewtoolkit.plugins.packagecurationproviders.api.PackageCurationProviderFactory
-import org.ossreviewtoolkit.utils.common.Options
 import org.ossreviewtoolkit.utils.ort.DeclaredLicenseProcessor
 import org.ossreviewtoolkit.utils.spdx.SpdxExpression
 
-class Sw360PackageCurationProviderFactory : PackageCurationProviderFactory<Sw360StorageConfiguration> {
-    override val type = "SW360"
-
-    override fun create(config: Sw360StorageConfiguration) = Sw360PackageCurationProvider(config)
-
-    override fun parseConfig(options: Options, secrets: Options) =
-        Sw360StorageConfiguration(
-            restUrl = options.getValue("restUrl"),
-            authUrl = options.getValue("authUrl"),
-            username = secrets.getValue("username"),
-            password = secrets["password"].orEmpty(),
-            clientId = secrets.getValue("clientId"),
-            clientPassword = secrets["clientPassword"].orEmpty(),
-            token = secrets["token"].orEmpty()
-        )
-}
+/**
+ * A class to hold the configuration for SW360.
+ */
+data class Sw360PackageCurationProviderConfig(
+    /**
+     * The REST API URL of SW360.
+     */
+    val restUrl: String,
+
+    /**
+     * The authentication URL of your SW360 instance.
+     */
+    val authUrl: String,
+
+    /**
+     * The username for the requests to SW360.
+     */
+    val username: Secret?,
```

Comment:
This is not nullable in the original code, why is it now?

And should usernames really be treated as secrets?

Reply:
I made it nullable because the docs of the `token` property say that it can be used instead of the other properties, but I haven't made `authUrl` nullable. What do you prefer, keep the value non-nullable as before or make `authUrl` nullable as well?

Reply:
For the sake of maintaining semantics as much as possible to not mix too many things in this change I'd prefer to keep values non-nullable as before.

Reply:
I have aligned nullability and default values with the previous implementation.

Reply:
> And should usernames really be treated as secrets?

This was discussed previously and there are arguments for both. However, it was read from the `secrets` map before so for this PR I'll keep it as it was.